crates/miden-objects/src/note/note_tag.rs

```diff
@@ -62,8 +66,19 @@ pub enum NoteExecutionMode {
 /// intended to be consumed by the network must be public to have all the details available. The
 /// public note for local execution is intended to allow users to search for notes that can be
```

Comment:
Let's also update the "Target" part of the docs (which I can't reference in this comment directly).

```
/// - Target describes how to further interpret the bits in the tag.
///   - For tags with an account target, the rest of the tag is interpreted as a partial
///     [`AccountId`]. For network accounts these are the first 30 bits of the ID while for local
///     account targets, the first 14 bits are used - a trade-off between privacy and uniqueness.
///   - For use case values, the meaning of the rest of the tag is not specified by the protocol and
///     can be used by applications built on top of the rollup.
```